keps/sig-storage/2644-honor-pv-reclaim-policy/README.md

```diff
@@ -0,0 +1,321 @@
+
+In the above scenario, the upgrade will cause change in default behavior as described in the current KEP. Additionally,
+if there are PVs that have a valid associated PVC and deletion timestamp set, then a finalizer is added to the PV.
+
+* Downgrade from new Kubernetes(1.23) to old Kubernetes(1.22).
+
+In this case, there may be PVs with the deletion finalizer that the older Kubernetes does not remove. Such PVs will be in the API server forever unless if the user manually removes them.
+
+### Version Skew Strategy
+The fix is part of `kube-controller-manager` of `external-provisioner`.
+
+1. `kube-controller-manager` is upgraded, but `external-provisioner` is not:
+
+In this case the changed behavior is observed only for in-tree plugins. The drivers would still have the issue. Users
+will observe the new annotation on the PV, but it does not have any effect since the `external-provisioner` is not
+updated
+
+2. `external-provisioner` is upgraded but `kube-controller-manager`
+
+In this case the annotations are not added, the upgraded `external-provisioner` will have no effect, the issue would
+still exist.
+
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: `HonorPVReclaimPolicy`
+  - Components depending on the feature gate: kube-controller-manager, external-provisioner
+
+###### Does enabling the feature change any default behavior?
+
+Enabling the feature will delete the volume from underlying storage when the PV is deleted followed deleting the PVC for
+a bound PV-PVC pair where the PV reclaim policy is delete.
+
+###### Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?
+Yes. Disabling the feature flag will continue previous behavior.
+
+###### What happens if we reenable the feature if it was previously rolled back?
+Will pick the new behavior as described before.
+
+###### Are there any tests for feature enablement/disablement?
+
+There will be unit tests for the feature `HonorPVReclaimPolicy` enablement/disablement.
+
+### Scalability
+
+###### Will enabling / using this feature result in any new API calls?
+No.
+
+###### Will enabling / using this feature result in introducing new API types?
+No.
+
+###### Will enabling / using this feature result in any new calls to the cloud provider?
+Yes, previously the delete volume call would not reach the provider under the described circumstances, with this change
+the volume delete will be invoked to remove the volume from the underlying storage.
+
+###### Will enabling / using this feature result in increasing size or count of the existing API objects?
+No.
+
+###### Will enabling / using this feature result in increasing time taken by any operations covered by existing SLIs/SLOs?
+No.
```

Comment:
Just curious, do we have SLOs for PV deletion?

Reply:
I don't know about that.  We have metric for delete volume operations.  So potentially the time to delete a PV could be longer now since we want to make sure volume on the storage system is deleted first before deleting the PV.  I'll add this.